server/src/test/java/io/crate/integrationtests/PostgresITest.java

```diff
@@ -645,8 +645,11 @@ public void testExecuteBatchWithOneFailingAndNothingExecuted() throws Exception
             preparedStatement.setString(1, Integer.toString(3));
             preparedStatement.addBatch();
 
-            assertThat(preparedStatement.executeBatch()).isEqualTo(new int[]{0, 0, 0});
+            assertThatThrownBy(preparedStatement::executeBatch)
+                .isExactlyInstanceOf(BatchUpdateException.class)
+                .hasMessageContaining("Cannot cast value `foobar` to type `integer`");
```

Comment:
I've verified this with PG17, it will also fail the complete `executeBatch` operation with an analysis conversion error.

Reply:
Can you also verify what is the result of `batchUpdateException.getUpdateCounts();`?
Should we return an array of `-3` (`Statement.EXECUTE_FAILED) ? (double check with postgres)
Also, should we have any special handling of the `nextException()` mechanism to repeat for the number of bulk args? (also need to check with postgres).